Feature - Foundation.MediaQuery.is() Method

auginator opened the pull request:
I was thinking this would be a good way for plugins to target a specific breakpoint, rather than being limited to a minimum breakpoint, as they are now with atLeast().  Accepts either ‘[breakpoint] only’ or ‘[breakpoint]’ - in the former case, it will see if the breakpoint matches exactly, and in the latter, it falls back to atLeast(). 

For instance: #8864 

@brettsmason @rafibomb  - Any thoughts on this? I would be happy to take a stab at integrating this into the plugins if the community thinks it is worthwhile. 

PS: It's my first attempt at contributing to Foundation - I hope I did not donk things up too badly!

auginator committed:
Added proposed public is() method for use in plugins

Accepts either ‘[breakpoint] only’ or ‘[breakpoint]’ - in the former
case, it will see if the breakpoint matches exactly, and in the latter,
it falls back to atLeast(). I was thinking this would be a good way for
plugins to target a specific breakpoint, rather than being limited to
atLeast.

rafibomb commented:
This is a cool feature - so if I read it right this will allow `data-equalize-on="medium only"`

rafibomb commented:
We'll pull it down and test it - thanks for pointing to 6.3!

auginator commented:
Nice – I would still have to change methods on the plugins that use Foundation.MediaQuery to use this function instead of the atLeast() method, but I am more than willing to give it a shot and update the pull request. 

Happy I got the version number for feature requests right!

auginator committed:
Example implimentation

– Fixed a derp in the mediaQuery is method’s conditional check for the
length of the string.
- Replaced atLeast() with new is() method in equalizer, tooltip and
sticky

auginator commented:
I updated the branch with a few changes :) Now it works as planned with Equalizer, Tooltip, and Sticky. Cheers! 

**Equalizer:** 
`<div class="row" data-equalizer data-equalize-on="medium only" id="the-equal">...</div>`

**Sticky:** 
`<h3 class="callout" data-sticky data-stick-to="bottom" data-sticky-on="medium only">I stick on medium only!</h3>`

**Tooltip:** *(I must admit this one seems a little weird to me, not sure if it would practically be that useful.)*
`<span data-tooltip data-show-on="small only" …>…</span>`

auginator commented:
@rafibomb I forgot to @ mention you when I updated the branch one week ago.

kball commented:
I like this.  Nice work.
